### DISCUSSION TITLE
BT-702(b)-notice error validation using XPath

### DISCUSSION THREAD

**Discussion by user6330** [OP-TED/eForms-SDK#430] [Q&A] [answered]
I encounterd this problems using EfxTranslator for the field **BT-702(b)-notice**.
I compare the field **BT-702(a)-notice** and I found out maybe the error on field.json.

`"id": "BT-702(a)-notice",
...
"assert": {
                    "value": "{ND-Root} ${TRUE}",
                    "severity": "ERROR",
                    "constraints": [{
                            "condition": "{ND-Root} ${BT-702(b)-notice is present}",
                            "value": "{ND-Root} ${not(BT-702(a)-notice in BT-702(b)-notice)}",
                            "severity": "ERROR",
                            "message": "rule|text|BR-BT-00702-0150"
                        }
                    ]
                }`

`"id": "BT-702(b)-notice".
.....
"assert": {
                    "value": "{ND-Root} ${TRUE}",
                    "severity": "ERROR",
                    "constraints": [{
                            "condition": "{BT-702(b)-notice} ${BT-702(b)-notice is present}",
                            "value": "{BT-702(b)-notice} ${BT-702(b)-notice is unique in /BT-702(b)-notice}",
                            "severity": "ERROR",
                            "message": "rule|text|BR-BT-00702-0151"
                        }
                    ]
                }`

The **BT-702(a)-notice** condition and value has as first param _{ND-Root}_, while the **BT-702(b)-notice** has _{BT-702(b)-notice}_.

In fact the efxtranslator, convert 
{
    "id": "BT-702(a)-notice",
    "condition": "{ND-Root} ${BT-702(b)-notice is present}",
    "translatedCondition": "//cac:AdditionalNoticeLanguage/cbc:ID",
    "value": "{ND-Root} ${not(BT-702(a)-notice in BT-702(b)-notice)}",
    "translatedValue": "not(//cbc:NoticeLanguageCode/normalize-space(text()) = //cac:AdditionalNoticeLanguage/cbc:ID)"
}

{
    "id": "BT-702(b)-notice",
    "condition": "{BT-702(b)-notice} ${BT-702(b)-notice is present}",
    "translatedCondition": "//.",
    "value": "{BT-702(b)-notice} ${BT-702(b)-notice is unique in /BT-702(b)-notice}",
    "translatedValue": "count(for $x in //., $y in //cac:AdditionalNoticeLanguage/cbc:ID[. = $x] return $y) = 1"
}

Can you please fix this case?

**Comment by YvesJo** [comment, marked as answer]
Hi,
There are indeed different contexts. 
The first rule is there to check that the Notice language specified in cbc:NoticeLanguageCode differs from any other specified notice languages (in cac:AdditionalNoticeLanguage/cbc:ID)
The second rule is there to check for any specified additional notice language (context {BT-702(b)-notice}), the code differs (i.e. it appears only once) from the codes of any other specified additional languages ( /BT-702(b)-notice})
